katdal/chunkstore_s3.py

```diff
@@ -70,16 +70,22 @@ def __init__(self, client):
         self.client = client
 
     @classmethod
-    def from_url(cls, url):
+    def from_url(cls, url, **kwargs):
         """Construct S3 chunk store from endpoint URL.
 
         S3 authentication (i.e. the access + secret keys) is handled externally
-        via the botocore config file or environment variables.
+        via the botocore config file or environment variables. Extra keyword
+        arguments are interpreted as botocore config settings (see
+        :class:`botocore.config.Config`) or arguments to the client creation
+        method (see :meth:`botocore.session.Session.create_client`), in that
+        order, overriding the defaults.
```

Comment:
I don't know what arguments are taken by each, but is there any risk that
(a) a current client argument is shadowed by a config argument and hence cannot be set through this interface?
(b) a future version of botocore would introduce a config argument that has the same name as a client argument, causing code that was expecting an argument to be passed to the client to be passed to the config instead?

Reply:
(a) The only common one is `region_name` which has the same meaning.
(b) The risk is low as the two is meant to complement each other. I'll worry about it if it happens. In that case one is still free to build a whole new custom botocore `Client` from scratch.